player has lifes and if he is hit by explosion, lifes decreases

Abi2737 opened the pull request.

MateiEne committed:
player has lifes and if he is hit by explosion, lifes decreases

Abi2737 commented on BasePlayer.cpp:
```diff
@@ -398,6 +401,39 @@ void BasePlayer::Stay()
+	if (lifes == 0)
+	{
+		OnDeath();
+		return;
```
nit: `return` not needed

MateiEne replied:
ok

Abi2737 commented on BasePlayer.cpp:
```diff
@@ -398,6 +401,39 @@ void BasePlayer::Stay()
+
+void BasePlayer::OnDeath()
+{
+	cout << name << " e mort" << endl;
```
nit: romengleza :P

MateiEne replied:
📦 :P

Abi2737 commented on BasePlayer.cpp:
```diff
@@ -398,6 +401,39 @@ void BasePlayer::Stay()
+void BasePlayer::OnDeath()
+{
+	cout << name << " e mort" << endl;
+	return;
```
nit: redundant `return`

Abi2737 commented on Bomb.cpp:
```diff
@@ -435,9 +435,8 @@ void Bomb::MarkExplosionXSideInMap(int lenght, bool right, char ch)
-	pos.l += 1 * sign;
 
-	while (lenght)
+	while (lenght >= 0)
```
nice catch ;) 👏

MateiEne replied:
haha :)) i m a genius :))

Abi2737 replied:
:))) i think you're also the guy who wrote these methods

MateiEne replied:
xD

Abi2737 commented on Constants.h:
```diff
@@ -39,6 +39,7 @@ namespace PlayerConst
 	constexpr float SPEED = 100;
 	constexpr int BOMB_COUNT = 3;
 	constexpr int BOMB_LENGTH = 3;
+	constexpr int LIFES = 3;
```
Thoughts  on naming it `LIFES_COUNT`?
And also, we could rename `BOX_COUNT` to `BOXES_COUNT` and `BOMB_COUNT` to `BOMBs_COUNT`

MateiEne replied:
done

Abi2737 commented on BasePlayer.h:
```diff
@@ -48,15 +50,19 @@ class BasePlayer
+	void HitBox();
 
 protected:
+	int lifes;
```
would you like more: `numOfLifes`?

MateiEne committed:
some variables name changes

MateiEne committed:
player is invincible for an amount of seconds if he is hit

MateiEne committed:
transparency when player is hit

MateiEne committed:
deadWalker is unkillable

MateiEne committed:
The player cannot move if is in life lost or dead animation;
If needed, the player moves to a good mat position after the life lost animation has ended.
Cannot put bomb if the player is killed or life lost animation is playing.

MateiEne committed:
modiefied explosion colors

Abi2737 committed:
fixed bug when the bomb explosion removes multiple boxes
improved bomb marking in map
improved length animination times

Abi2737 committed:
Merge branch 'master' into player-explosion-hitbox
